core/src/test/java/com/linecorp/armeria/server/websocket/WebSocketIdleTimeoutTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.websocket;
+
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.websocket.WebSocketClient;
+import com.linecorp.armeria.client.websocket.WebSocketSession;
+import com.linecorp.armeria.common.ClosedSessionException;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.websocket.WebSocketWriter;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.websocket.WebSocketServiceTest.AbstractWebSocketHandler;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+public class WebSocketIdleTimeoutTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            final WebSocketService service =
+                    WebSocketService.builder(new AbstractWebSocketHandler()).build();
+            sb.service("/idle", service);
+            sb.idleTimeout(Duration.ofMillis(1));
+        }
+    };
+
+    @Test
+    void shouldThrowWhenWritingToIdleWebSocket() throws Exception {
+        final Throwable ex = assertThrows(RuntimeException.class, () -> {
+            final WebSocketClient client = WebSocketClient.of(SessionProtocol.H1C, server.httpEndpoint());
```

Comment:
@ikhoon Did you imagine that only HTTP1 needs this kind of behavior when creating the issue? I'm curious if we don't need to consider H2C as well.

Reply:
- Thank you for your quick review. 
- In case of a failed test, it may be a little late because I'm in the military. But I'll fix it ASAP.
- Below is a summary of why I didn't consider H2C, and what I understood about the WebSocket code. 
- Please let me know if there's anything wrong.

## Reason
1. In HTTP2 protocol, close the idle channel by the ```AbstractKeepAliveHandler``` class, the parent of the ```HTTP2ServerKeepAliveHandler``` generated by the ```HTTP2ServerConnectionHandler``` 

2. HTTP2 idleTimeout features already tested in ```KeepAliveHandlerTest``` 

3. I modified, the ```WebSocketServiceChannelHandler```, is registered in Channel Pipeline ```only for HTTP1 protocol``` 

4. ```AbstractKeepAliveHandler``` was using ```idleTimeoutMillis``` in ```ServerConfig``` to close the idle channel, ```WebSocketServiceChannelHandler``` also uses ```idleTimeoutMillis``` in ```ServerConfig``` for the same result


## About Web Socket Operation

### HTTP/2(H2C)
1. When connected by ```SessionProtocol.H2C```, the ```Http2ServerConnectionHandler``` is registered in the channel pipeline as shown in the picture. 

![image](https://github.com/line/armeria/assets/34996184/9afa88e4-1a2f-421c-8b36-562204542470)

2. When the ```Http2ServerConnectionHandler``` is created, if the ```idleTimeoutMillis``` is not ```0```,  create the ```Http2ServerKeepAliveHandler``` as shown in the picture below.

![image](https://github.com/line/armeria/assets/34996184/127d5542-1958-45eb-95f7-0841db6884ac)

3. In ```AbstractKeepAliveHandler```, parent of the ```Http2ServerKeepAliveHandler```,  close the idle channel like the log below.

```
16:20:27.830 [armeria-common-worker-nio-3-2] DEBUG c.l.a.i.c.AbstractKeepAliveHandler - [id: 0x0af5558b, L:/127.0.0.1:55809 - R:/127.0.0.1:55811] Closing an idle server connection
```

### HTTP/1(H1C)
1. When you connect to ```SessionProtocol.H1C```, decode the request in the ```Http1RequestDecoder```.

![image](https://github.com/line/armeria/assets/34996184/f8d5c12e-b36e-4e96-9640-c47b16b5f7f4)

2. If the request is WebSocketUpgradeRequest, replace the ```Http1RequestDecoder``` with the ```WebSocketServiceChannelHandler``` as shown in the picture below.

![image](https://github.com/line/armeria/assets/34996184/220d9b32-ad94-46e5-a76b-40b919ada3e6)

3. However, there was no idleTimeout-related code in the existing WebSocketServiceChannelHandler 

4. So when connected with the HTTP1 protocol, the channel did not close if the idle state continued.

Reply:
> In HTTP2 protocol, close the idle channel by the AbstractKeepAliveHandler class, the parent of the HTTP2ServerKeepAliveHandler generated by the HTTP2ServerConnectionHandler

Sure, but I thought `Http2ServerKeepAliveHandler` also operated on the connection level. I was asking the original issue creator because I was looking at the following line which seemed to imply an idle timeout for requests which h2c doesn't have at the moment.

> The existing idle timeout handler operates at the connection level so a new scheduler that can perform at the WebSocket level is necessary.